pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -163,6 +175,13 @@ type AzVolumeAttachmentList struct {
 	Items []AzVolumeAttachment `json:"items"`
 }
 
+type AzError struct {
+	ErrorCode    string            `json:"errorCode"`
+	ErrorMessage string            `json:"errorMessage"`
+	CurrentNode  k8stypes.NodeName `json:"currentNode"`
```

Comment:
Rather than have error-specific fields added to the generic `AzError` class, I wonder if we should create a specific error class for the dangling attach error that embeds `AzError`, e.g.

```go
type AzDanglingAttachError struct {
    AzError
    CurrentNode k8stypes.NodeName
    DevicePath string
}
```

Reply:
NM. I don't think this will work. Not sure that JSON marshaling handles polymorphism.

Reply:
I don't think it does unfortunately. I have tried more general error approach but crd generation gave me failures.